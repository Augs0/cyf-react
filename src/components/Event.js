import React from 'react';

export const Event = (event) => {

  const {
    id,
    time,
    date,
    title,
    location,
    mentors,
    sponsors
  } = event;

  return (
    <ul className="no-bullet">
      <div className="row grouped_events">
        <div className="medium-12 columns">
          <h4>{date}</h4>
        </div>
        <div className='event'>
          <div className='row'>
            <div className='small-3 medium-2 columns'>
              <div className='date'>
                <div className='time'>
                {time}
                </div>
              </div>
            <div className='text-center'>
              <label className='label info'>
                {location}
              </label>
            </div>
          </div>
        </div>
        <div className="small-9 medium-10 columns">
          <div className="title">
            <a href={`/events/${id}`}>{title}</a>
            &nbsp;at&nbsp;
            <a href="">{location}</a>
          </div>
          <div className="row">
            <div className="medium-6 columns">
              <div className="sponsors">
                {sponsors[0].name}
              </div>
            </div>
            <div className="medium-6 columns">
              <div className="mentors">
                {mentors[0].name}
              </div>
            </div>
          </div>
    </div>
    </div>
    </div>
  </ul>
  )
}