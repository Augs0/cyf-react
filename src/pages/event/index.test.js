import React from 'react';
import { shallow } from 'enzyme';

import Event from './';

describe('Event Component', () => {
  it('renders', () => {
    expect(shallow(<Event />)).toMatchSnapshot();
  });
});
