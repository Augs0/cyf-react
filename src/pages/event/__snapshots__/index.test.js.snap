// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component renders 1`] = `
<Styled(div)>
  <Styled(div)>
    <EventDetail
      city="London"
      description="First bit of javascript"
      endDate="1515851599504"
      intake="London Class 3"
      links={
        Array [
          Object {
            "course materials": "https://codeyourfuture.github.io/syllabus-master/js-core/week-04/lesson.html",
          },
        ]
      }
      location="Ticketmaster"
      mentors={
        Array [
          Object {
            "avatar": "www.something.com",
            "name": "Matt",
          },
          Object {
            "avatar": "www.somthing.com",
            "name": "Jun",
          },
          Object {
            "avatar": "www.somthing.com",
            "name": "Felix",
          },
        ]
      }
      moduleLeaders={
        Array [
          Object {
            "avatar": "something",
            "name": "Kash",
          },
        ]
      }
      startDate="1515851599504"
      topic="Week 4 - Hello Javascript"
    />
  </Styled(div)>
</Styled(div)>
`;
